### PR TITLE
tests: sort metrics by type in multiTrialSample

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

DET-9608

## Test Plan
```
hbp ➜ master git:(trialsample-test) gotestsum ./internal --count 1000 --run TestMultiTrialSample
✓  internal (2m12.193s)

DONE 1000 tests in 133.869s
```

## Commentary (optional)

the main fix is[ this commit](https://github.com/determined-ai/determined/pull/7191/commits/ed99a2447863f554ce9b3cc63d9b03a2de61567e) 

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by hamidzr** [comment]
I tried changing the branches hash to get ci to run theses tests to no avail

File: master/pkg/model/metrics_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+			false,
+		},
+	}
+	for idx, tt := range tests {
+		t.Run(fmt.Sprint(idx), func(t *testing.T) {
+			got, err := DeserializeMetricIdentifier(tt.args.s)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("DeserializeMetricIdentifier(%v) error = %v, wantErr %v", tt.args, err, tt.wantErr)
+				return
+			}
```
Comment (maxrussell) [inline master/pkg/model/metrics_test.go:71]: Nit: personally, I'm a fan of just using `if`s and stuff like this in testing, but if we're doing the whole assertion thing, I think we should keep it consistent.

```suggestion
			if tt.wantErr {
				require.Error(t, err, "Expected error with arg %v", tt.args.s)
			} else {
				require.NoError(t, err, "Unexpected error with arg %v", tt.args.s)
			}
```

**Review by maxrussell — APPROVED** [review]
LGTM, with a couple nits and the optional addition of a `MetricName` type.

File: master/pkg/model/metrics.go [hunk 2]
```diff
@@ -16,6 +17,15 @@ const (
 	TrainingMetricType MetricType = "training"
 )
 
+// ValidateMetricName validates the metric name.
+// CHAT: do we make a new type similar to MetricType?
```
[the remark this answers is not in the document]
Comment (stoksc) [reply master/pkg/model/metrics.go:21]: I do agree with @maxrussell 's feedback. Also, nit, this can be unexported.

**Comment by hamidzr** [comment]
https://github.com/determined-ai/determined/pull/7210#issuecomment-1608141325

**Pull request closed without merging**